shared/src/proxies/requests.ts

```diff
@@ -122,6 +125,43 @@ export const post = async ({
   }
 };
 
+export const asyncSyncHandler = (
+  applicationContext,
+  request,
+  asyncSyncId = applicationContext.getUniqueId(),
+) => {
+  getMemoized.clear();
+
+  return new Promise((resolve, reject) => {
+    const callback = results => {
+      applicationContext
+        .getAsynSyncUtil()
+        .removeAsyncSyncCompleter(asyncSyncId);
```

Comment:
Love the move to polling as a way of getting around websockets. I talked with Jon a little and I think it would be possible to achieve polling without using the asyncSyncCompleter Dict. I am suggesting not using the dictionary to reduce some of the steps in making polling happen and so we do not have an object in applicationContext that is holding many callback functions and growing in size accidentally.